src/i18n.mjs

```diff
@@ -74,7 +74,16 @@ class I18n {
   }
 
   setLocale (locale) {
-    this.currentLocale = locale
+
+    // Allow validation and re-writing of the local string
+    const candidate = this.whenChangingLocale(locale)
+
+    // Do not update locale if new value is null or undefined
+    if (candidate !== undefined && candidate !== null) {
```

Comment:
This can be simplied to:

```suggestion
if (candidate) {
```

Reply:
I have to disagree. Your suggestion would exclude the number 0, the `false` value and the empty string, while my code only excludes `null` and `undefined`. Actually, the current code is equivalent to `if (candidate == undefined) { ...` or `if (candidate == null) { ...` (since `null == undefined`) but I thought that explicit was better than implicit.

This point was addressed in [this question](https://github.com/ficusjs/ficusjs-i18n/issues/55#issuecomment-1652182244). This also ties into your next suggestion below, but even assuming that only `string` values are passed, `if (candidate) { ...` would still exclude the empty string in addition to `null` and `undefined`.

If ultimately this is what you prefer, I'm happy to change the code, but they're not equivalent.

Reply:
Ok, I see your point. If, as agreed, `null` or `undefined` are passed, the check can be simply `if (candidate != null) {` which checks for both `null` and `undefined` values.